### ISSUE TITLE
Enhance the FAQ with questions from RMIT projects

### ISSUE THREAD

**Issue by gallandarakhneorg** [sarl/sarl#855]
The FAQ provided by RMIT may be put into the official SARL FAQ:

https://bitbucket.org/snippets/ssardina/9er67X

**Comment by ssardina** [comment]
That's totally fine with me if you think it is useful! Go for it! I keep adding stuff there..

**Comment by gallandarakhneorg** [comment]
Additionally, the currnet FAQ of SARL must be reviewed in order to detect invalid context. This should not append because of the unit test of the documentation.

**Comment by gallandarakhneorg** [comment]
@alexandrelombard, please do not miss to add @ssardina has contributor to SARL into the POM file.

**Comment by ssardina** [comment]
Do you mean to add it as a link in Section 2 [here](http://www.sarl.io/docs/official/index.html)? That sounds good. I have actually been adding to it lately... :-)

**Comment by gallandarakhneorg** [comment]
I think we will extract the most import elements of your FAQ in order to make it directly accessible from the SARL FAQ (and put you as author of course). The rest of your FAQ will be reachable from a link.

**Comment by ssardina** [comment]
Perfect!

**Comment by gallandarakhneorg** [comment]
Dear @ssardina.
Your FAQ could be updated according to the progresses of version 0.9 of SARL.
Several questions are now obsolete.

**Comment by ssardina** [comment]
I will do that once 0.9.0 is officially released, but how do I know which ones are obsolete?

I know this one is obsolete:

I cannot pass data to the Initialize of a Behaivor, occurrence.paramters is always empty!

**Comment by gallandarakhneorg** [comment]
Dear @ssardina.
The changes to apply on your own page are:
* ***Strange error in CLI when using occurence on left hand side of assignments***: now there is an error if the SARL compiler has detected an assignment to a field of occurrence. Passing a field of occurrence to a function becomes a warning (See FAQ 0.9).
* ***Can we document SARL code for JAVADOC?***: yes (additionnally, in later version, I will provide a sarldoc command-line tool for helping).
* ***How do I control the log-level of the Logging built-in capacity?***: FAQ 0.9 gives additional details.
* ***Equality and identity comparison (==, ===, !=, !==) in SARL and checking for null: same as Java ?***: included into FAQ 0.9
* ***How to return two values? Pairs in SARL***: included into FAQ 0.9.
* ***I cannot pass data to the Initialize of a Behaivor, occurrence.paramters is always empty!***: not any more an issue in version 0.9.
* ***SARL is giving lots of WARNINGS at compile time, how can I avoid them?***: included into FAQ 0.9
* ***I cannot define a static field in an agent tye declaration (agent, skill, behavior), why?***: included into FAQ 0.9.
* ***Emitting events and spawning agents in "on Intialize" behaviors: be careful!***: included into FAQ 0.9.

By the way, you are now an official contributor to SARL as I push you as author of the commit related to this issue.

**Comment by ssardina** [comment]
Hi @gallandarakhneorg .

Great, I have checked each of them and all good; have simplified my FAQ for all of them.

I found one entry that had itemization wrongly formatted:

http://www.sarl.io/docs/official/faq/SyntaxFAQ.html#1-13-equality-and-identity-comparison-in-sarl-and-checking-for-null-same-as-java

So I fixed it and submitted a PR for that. I hope I did all correctly. Amazing the whole setup, very impressed.
